lib/common/mobile/mobile-core/devices-service.ts

```diff
@@ -287,51 +289,63 @@ export class DevicesService extends EventEmitter implements Mobile.IDevicesServi
-			this.$logger.trace("Device detection intervals are already started. New intervals will not be started.");
-			return;
-		}
+	@exported("devicesService")
+	public startDeviceDetectionInterval(deviceInitOpts: Mobile.IDeviceLookingOptions = <any>{}): void {
+		if (!this.deviceDetectionInterval) {
+			let isDeviceDetectionIntervalInProgress = false;
 
-		let isDeviceDetectionIntervalInProgress = false;
-		const deviceDetectionInterval = setInterval(async () => {
-			if (isDeviceDetectionIntervalInProgress) {
-				return;
-			}
+			this.deviceDetectionInterval = this.$timers.setInterval(async () => {
+				if (isDeviceDetectionIntervalInProgress) {
+					return;
+				}
 
-			isDeviceDetectionIntervalInProgress = true;
+				isDeviceDetectionIntervalInProgress = true;
 
-			await this.detectCurrentlyAttachedDevices(deviceInitOpts);
+				await this.detectCurrentlyAttachedDevices(deviceInitOpts);
 
-			try {
-				const trustedDevices = _.filter(this._devices, device => device.deviceInfo.status === constants.CONNECTED_STATUS);
-				await settlePromises(_.map(trustedDevices, device => device.applicationManager.checkForApplicationUpdates()));
-			} catch (err) {
-				this.$logger.trace("Error checking for application updates on devices.", err);
-			}
+				try {
+					const trustedDevices = _.filter(this._devices, device => device.deviceInfo.status === constants.CONNECTED_STATUS);
+					await settlePromises(_.map(trustedDevices, device => device.applicationManager.checkForApplicationUpdates()));
+				} catch (err) {
+					this.$logger.trace("Error checking for application updates on devices.", err);
+				}
 
-			isDeviceDetectionIntervalInProgress = false;
+				isDeviceDetectionIntervalInProgress = false;
 
-		}, DevicesService.DEVICE_LOOKING_INTERVAL);
+			}, deviceInitOpts.detectionInterval || DevicesService.DEVICE_LOOKING_INTERVAL);
 
-		deviceDetectionInterval.unref();
-		this.deviceDetectionIntervals.push(deviceDetectionInterval);
+			this.deviceDetectionInterval.unref();
+		}
+	}
 
+	@exported("devicesService")
+	public startEmulatorDetectionInterval(opts: Mobile.IHasDetectionInterval = {}): void {
 		let isEmulatorDetectionIntervalRunning = false;
-		const emulatorDetectionInterval = setInterval(async () => {
+		this.emulatorDetectionInterval = this.$timers.setInterval(async () => {
 			if (isEmulatorDetectionIntervalRunning) {
 				return;
 			}
 
 			isEmulatorDetectionIntervalRunning = true;
 			await this.detectCurrentlyAvailableEmulators();
 			isEmulatorDetectionIntervalRunning = false;
-		}, DevicesService.EMULATOR_IMAGES_DETECTION_INTERVAL);
+		}, opts.detectionInterval || DevicesService.EMULATOR_IMAGES_DETECTION_INTERVAL);
 
-		emulatorDetectionInterval.unref();
-		this.deviceDetectionIntervals.push(emulatorDetectionInterval);
+		this.emulatorDetectionInterval.unref();
+	}
+
+	@exported("devicesService")
+	public stopDeviceDetectionInterval(): void {
+		if (this.deviceDetectionInterval) {
+			clearInterval(this.deviceDetectionInterval);
```

Comment:
Maybe it'll be better to expose a public `clearInterval` method from `timers` service

Reply:
Done in latest commit.